includes/class-wc-customer-download-log.php

```diff
@@ -0,0 +1,242 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class for customer download logs.
+ *
+ * @version		3.3.0
+ * @since		3.3.0
+ * @package		WooCommerce/Classes
+ * @author		WooThemes
+ */
+class WC_Customer_Download_Log extends WC_Data implements ArrayAccess {
```

Comment:
Since this is a new class, why does it need ArrayAccess and backwards compatibility? Can you explain what this is replacing in the old codebase?